### DISCUSSION TITLE
Specifying image size for viewing in latex and markdown

### DISCUSSION THREAD

**Discussion by likandia** [Witiko/markdown#395] [Q&A] [answered]
Can an image have its width specified, and be able to be viewed in both markdown (say, Github markdown) and latex?

The following works to display the image in latex, but does not work with pure markdown. 

`![](img/example.png){#id .class width=400}`

And I believe you cannot use HTML tags to specify size to display the image within latex. Am I missing any options?

**Comment by Witiko** [comment, marked as answer]
Hi, thanks for using the Markdown package for TeX.

> Can an image have its width specified, and be able to be viewed in both markdown (say, Github markdown) and latex?
>
> The following works to display the image in latex, but does not work with pure markdown.
> 
> ``` markdown
> ![](img/example.png){#id .class width=400}
> ```

[Github-Flavored Markdown][6] does not support attributes on images. Therefore, you would not be able to use the code you mentioned.

Other flavors of Markdown, such as the one recognized by Pandoc with the [`link_attributes` extension][1] enabled (see the corresponding [option `linkAttributes`][2] for the Markdown package), would allow you to specify attributes on images. However, this would not solve your issue immediately, because the default LaTeX renderer for attributes on images passes the attributes directly to the `\includegraphics` command, which expects the width in TeX's units such as `width=5mm`, whereas Pandoc expects the width either in pixels or as percentage. Therefore, you would need to redefine the LaTeX renderer to support width specified in pixels (by assuming a certain DPI) or as percentage. In the future, we may want to support these by default.

> And I believe you cannot use HTML tags to specify size to display the image within latex.

You might use HTML tags. When you enable the [option `html`][3] in the Markdown package, inline HTML markup such as `<img src="image.png" width="400">` will produce the [`inlineHtmlTag` renderer][4], which will receive the text of the HTML tag as its first argument. However, you would need to write the logic for parsing the inline HTML tag using e.g. [the l3regex library of the expl3 programming language for TeX][5]. This is not difficult if you are familiar with expl3. Please, let me know if you'd like some help with this.

 [1]: https://pandoc.org/MANUAL.html#extension-link_attributes
 [2]: https://witiko.github.io/markdown/#option-linkattributes
 [3]: https://witiko.github.io/markdown/#option-html
 [4]: https://witiko.github.io/markdown/#html-tag-and-element-renderers
 [5]: http://mirrors.ctan.org/macros/latex/contrib/l3kernel/interface3.pdf
 [6]: https://github.github.com/gfm/

**Reply by likandia** [reply]
Thanks! I'll give it a try
